File: backend/backend/src/main/java/com/backend/backend/dto/JobDto.java
package com.backend.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JobDto {
    private Long id;
    private String title;
    private String city;
    private String school;
    private Double minSalary;
    private Double maxSalary;
    private String type; // FULL_TIME or PART_TIME
    private Integer hours;
    private String workMode; // ONSITE, REMOTE, or HYBRID
}

File: frontend/src/dto/job-dto.ts
import {WorkMode} from '../enums/work-mode.enum';
import {JobType} from '../enums/job-type.enum';


export class JobDto {
  title: string;
  city: string;
  school: string;
  minSalary: number;
  maxSalary: number;
  type: JobType; // FULL_TIME or PART_TIME
  hours: number;
  workMode: WorkMode; // ONSITE, REMOTE, or HYBRID

  constructor(
    title: string,
    city: string,
    school: string,
    minSalary: number,
    maxSalary: number,
    type: JobType,
    hours: number,
    workMode: WorkMode
  ) {
    this.title = title;
    this.city = city;
    this.school = school;
    this.minSalary = minSalary;
    this.maxSalary = maxSalary;
    this.type = type;
    this.hours = hours;
    this.workMode = workMode;
  }
}
